lib/cocoapods/generator/xcconfig/xcconfig_helper.rb

```diff
@@ -278,7 +281,7 @@ def self.settings_for_dependent_targets(target, dependent_targets)
           }
 
           # Scope pod targets
-          if target.respond_to?(:configuration_build_dir)
+          if !test_xcconfig && target.respond_to?(:configuration_build_dir)
```

Comment:
`CONFIGURATION_BUILD_DIR` should not be set for targets otherwise the `.sh` scripts fail to execute. This is the same for user targets.